src/components/App.js

```diff
@@ -1,17 +1,94 @@
-import React from 'react';
+import React from "react";
+import Search from "./Search.js";
+import Films from "./Films.js";
+import FilmImage from "./FilmImage.js";
+import Favourites from "./Favourites.js";
+
+//`http://www.omdbapi.com/?s=${film}&apikey=d213a5b1`
+//`http://www.omdbapi.com/?i=${singleFilmInfo}&apikey=d213a5b1`
 
 class App extends React.Component {
-  constructor(){
+  constructor() {
     super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.removeFavouriteByIndex = this.removeFavouriteByIndex.bind(this);
+
+    this.state = {
+      filmSearch: "",
+      results: [],
+      favouriteFilm: []
+    };
+  }
+
+  componentDidMount() {}
+
+  handleClick(filmTitle) {
+    console.log(filmTitle);
+
+    const favouritesCopy = this.state.favouriteFilm.splice(0);
+    favouritesCopy.push(filmTitle);
+
+    this.setState({ favouriteFilm: favouritesCopy });
+  }
+
+  handleChange(event) {
+    event.preventDefault();
```

Comment:
rather than passing up events from Search to App, it would better to extract `event.target.value` in Search and just send up the string. It makes life slightly easier as events get dealt with in the component they are created in and only the relevant values are passed up.